test/plugin_functional/plugins/core_plugin_a/public/plugin.tsx

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { PluginInitializer, PluginSetupContext } from 'kibana/public';
+
+export class CorePluginAPlugin implements ReturnType<PluginInitializer<CorePluginAPluginSetup>> {
```

Comment:
optional nit: tests are documentation as well and can be used as a reference to recommended practices.
wouldn't it be a bit more user-friendly if we introduce a helper or declare plugin api explicitly? for example
```typescript
type PluginPublicApi<T> = ReturnType<PluginInitializer<T>>
class CorePluginAPlugin implements PluginPublicApi<CorePluginAPluginSetup> {...
//...or
interface PluginA { setup(): ...
class CorePluginAPlugin implements PluginA {...
```

Reply:
Looks like you're viewing an outdated version of the PR? I added an interface like this in a commit I pushed yesterday :)

Reply:
lol. I reviewed commit-by-commit and stopped after the first one because had to go out :)